perfaware/sim86/shared/Sim8086.cpp

```diff
@@ -303,62 +336,72 @@ int main(int ArgCount, char** Args)
     instruction_table Table;
     Sim86_Get8086InstructionTable(&Table);
     printf("8086 Instruction Instruction Encoding Count: %u\n", Table.EncodingCount);
-  
-    if (ArgCount > 1)
-    { 
-        for (int ArgIndex = 1; ArgIndex < ArgCount; ArgIndex++)
-        {
-            s16 Registers[RegisterCount] = {};
-            bool FlagArray[Flag_count] = {};
 
-            std::string OutputBuffer;
-            char* FileName = Args[ArgIndex];
-            std::ifstream File;
-            File.open(FileName, std::ifstream::binary | std::ifstream::in);
-            if (!File.good())
-            {
-                std::cout << "Error opening file " << FileName;
-                return -1;
-            }
-            
-            std::cout << "\n" << FileName << "\n";
-            std::vector<u8> Bytes((std::istreambuf_iterator<char>(File)), std::istreambuf_iterator<char>());
-            const u16 BytesRead = static_cast<u16>(Bytes.size());
-            u16& InstructionPointer = reinterpret_cast<u16&>(Registers[IPRegister]);
+    Memory = reinterpret_cast<u8*>(malloc(Megabyte));
 
-            while (InstructionPointer < BytesRead)
+    if (Memory)
+    {
+        if (ArgCount > 1)
```

Comment:
Instead of nesting your code like this, a nice alternative is to test the opposite condition and just return if it fails. This way the rest of your code doesn't have to be indented, and people who read don't have to hunt for the matching brace to see what exactly is contained in these large if statements.

Reply:
Fixed after removing runtime allocation
3446096e8be08fed3c8096a1c73251505b2c1509

Reply:
You still do the if and then a large block with all the code inside, and then return if you don't enter in the if. That's the same as before. The alternative I was talking about is to do the if with the opposite condition, and immediately return if needed. So all the code that's in the if now will be outside, and it won't require indentation.

But now, that's even a moot point, because you can just remove the if altogether, it's not needed at all.

Reply:
The same concept applies to the if (ArgCount >1) condition though, so I used it
7e442ff180e1912b993a76c1677c69d94e9d946d

Reply:
Good! But you can remove the if completely as I was saying before. You already check the same condition in the for loop.

Reply:
Whoops
e82f5524bdfe44589cedad54f5352e541c050b6e